src/typeInference/annotateBinaryOperator.ts

```diff
@@ -36,7 +36,11 @@ export function annotateBinOp(
 	}
 
 	// TODO: Fix this hack (pathExpr returns a node in 1 case, which cannot be added to an integer)
-	if (left.type === ValueType.NODE || right.type === ValueType.NODE) return undefined;
+	if (left.type === ValueType.NODE || right.type === ValueType.NODE)
```

Comment:
This should happen instead of throwing an error, these if checks were temporary fixes and should be removed.